src/lib/token_list_gen.ts

```diff
@@ -63,48 +63,56 @@ export const generateTokenList = async (
   const name = l1TokenList.name
   const mainLogoUri = l1TokenList.logoURI
 
-  const bridgeData = await instantiateBridge();
-  const { bridge, l1Network, l2Network } = bridgeData;
+  const { l1 , l2 } = await getNetworkConfig();
+
   let tokens =
     options && options.getAllTokensInNetwork
-      ? await getAllTokens(l2Network.chainID)
+      ? await getAllTokens(l2.network.chainID)
       : await getTokens(
           l1TokenList.tokens.map((token) => token.address.toLowerCase()),
-          l2Network.chainID
+          l2.network.chainID
         );
 
   
   const l1TokenAddresses = tokens.map((token:GraphTokenResult) => token.l1TokenAddr);
-  const l2AddressesFromL1 = await getL2TokenAddressesFromL1(l1TokenAddresses, bridge);
-  const l2AddressesFromL2 = await getL2TokenAddressesFromL2(l1TokenAddresses, bridge);
+  const l2AddressesFromL1 = await getL2TokenAddressesFromL1(l1TokenAddresses, l1.multiCaller, l2.network.tokenBridge.l1GatewayRouter);
+  const l2AddressesFromL2 = await getL2TokenAddressesFromL2(l1TokenAddresses, l2.multiCaller, l2.network.tokenBridge.l2GatewayRouter);
 
   // if the l2 route hasn't been updated yet we remove the token from the bridged tokens
   tokens = tokens.filter((t, i) => l2AddressesFromL1[i] === l2AddressesFromL2[i])
 
-  const tokenData = await getL2TokenData(l2AddressesFromL1, bridge);
+  const tokenData = await l2.multiCaller.getTokenData(
+    l2AddressesFromL1.map(t => t || constants.AddressZero),
+    { name: true, decimals: true, symbol: true }
+  )
   const logoUris: (string | undefined)[] = [];
   for (const token of tokens) {
     const uri = await getLogoUri(token.l1TokenAddr);
     logoUris.push(uri);
   }
 
-  let arbifiedTokenList:ArbTokenInfo[] = tokens.map((token, i: number) => {
-    const l2GatewayAddress = token.joinTableEntry[0].gateway.gatewayAddr;
-    const address = l2AddressesFromL1[i];
-    let { name:_name, decimals, symbol:_symbol } = tokenData[i];
+  let arbifiedTokenList:ArbTokenInfo[] = tokens
+      .map((t, i) => ({token: t, l2Address: l2AddressesFromL2[i], tokenDatum: tokenData[i]}))
+      .filter((t): t is typeof t & { l2Address: string } => t.l2Address != undefined && t.l2Address !== constants.AddressZero)
```

Comment:
I might be misreading it, but isn't this filter redundant with the above

 ```tokens = tokens.filter((t, i) => l2AddressesFromL1[i] === l2AddressesFromL2[i])```
? 

Maybe add some comments here broadly explaining the map/filter/map levels? (Even if only to help me 😉 )

Reply:
I found that we got the zero address there sometimes, I guess it means that our graphql query may return some token addresses that havent been fully bridged. Maybe it's a syncing issue in the graph?